src/io/tests/test_hdf_manual_reorder.cpp

```diff
@@ -0,0 +1,99 @@
+
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2018 Jeongnim Kim and QMCPACK developers.
+//
+// File developed by: Luke Shulenburger, lshulen@sandia.gov, Sandia National Laboratories
+//
+// File created by: Luke Shulenburger, lshulen@sandia.gov, Sandia National Laboratories
+//////////////////////////////////////////////////////////////////////////////////////
+
+
+#include "catch.hpp"
+#include "io/hdf_archive.h"
+#include <iostream>
+#include <vector>
+
+using std::cout;
+using std::endl;
+using std::vector;
+using namespace qmcplusplus;
+
+
+TEST_CASE("hdf_write_reorder", "[hdf]")
+{
+  hdf_archive hd;
+  hd.create("test_write_reorder.hdf");
+
+  vector<double> v{1.1, -2.2, 3.3, -4.4, 5.5, -6.6, 7.7, -8.8, 9.9};
+  vector<hsize_t> dims{3,3};
```

Comment:
Would the test be more powerful if the dimensions were not square?   I would be concerned about getting the dimensions reversed at some point in the code path.

Reply:
I looked more closely and see the following test handles the non-square case while reading.